### ISSUE TITLE
Inconsitency with dictonary members

### ISSUE THREAD

**Issue by crakjie** [graalvm/graalpython#56]
the following code call a lambda which return a python dictionary.
```
val result = calculator.execute(columns,lines)
println("result: "+result)
println("hasMembers: "+result.hasMembers)
println("hasMember(\"Col1\"): "+result.hasMember("Col1"))
println("hasMember(\"0\"): "+result.hasMember("0"))
println("getMemberKeys: "+result.getMemberKeys)
println("getMember(\"Col1\"): "+result.getMember("Col1"))
println("getMember(\"Col1\").hasMembers: "+result.getMember("Col1").hasMembers)
println("getMember(\"Col1\").getMemberKeys: "+result.getMember("Col1").getMemberKeys)
println("getMember(\"Col1\").getMember(\"0\"): "+result.getMember("Col1").getMember("0"))
```

the result of the execution.
```
result: {'Col1': {0: 1, 1: 4, 2: 7}, 'Col2': {0: 2, 1: 5, 2: 8}, 'Col3': {0: 3, 1: 6, 2: 9}}
hasMembers: true
hasMember("Col1"): false
hasMember("0"): false
getMemberKeys: [pop, fromkeys, setdefault, keys, values, get, clear, update, popitem, copy, items]
getMember("Col1"): {0: 1, 1: 4, 2: 7}
getMember("Col1").hasMembers: true
getMember("Col1").getMemberKeys: [pop, fromkeys, setdefault, keys, values, get, clear, update, popitem, copy, items]
getMember("Col1").getMember("0"): null
```

`hasMember("Col1")` return `false` but `getMember("Col1")` return the inner object.
`getMemberKeys` don't contain `"Col1"`
`getMember("Col1").getMember("0")` return null because the inner key is an `Int` and not a `String`

**Comment by timfel** [comment]
This is intended behaviour. We will soon change how members work, but for the time being, dictionary members that are not strings will not be exposed. The members returns the attributes of an object, regardless of the type object. The only way to get at the keys of a Python dictionary is to invoke the `keys` message, however, this will give you a Python iterable that you cannot do anything with from Java. To get at a dictionary value, you have to invoke the `get` method.

**Comment by timfel** [comment]
There's a bit of documentation about this in https://github.com/graalvm/graalpython/blob/master/doc/INTEROP.md

**Comment by timfel** [comment]
Maybe a little more info about upcoming changes to this: we will at a later stage distinguish "members" from "keys" or "contents", but at this stage we cannot. Since the limitation of the API also doesn't allow us to have non-string members, the only option for now really is to expose the methods and let the Java embedder use the Python methods to get at the contents of a dictionary. When using the `getMember` function, we try to be a little bit clever and if there's no such member but there is a key, we try to return it. But this cleverness does really only get us so far.

tl;dr: For now use the Python methods, this will work now and in the future. For more convenience, there will be some form of "member namespaces" in the future.
